### PR TITLE
Correctly set first_run_date as a datetime metric

### PR DESCRIPTION
**Pull request by badboy**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: glean-core/src/lib.rs [hunk 1]
```diff
@@ -72,9 +72,7 @@ impl Glean {
 
     fn initialize_core_metrics(&mut self) -> Result<()> {
         if first_run::is_first_run(&self.data_path)? {
```
Comment (mdboom) [inline glean-core/src/lib.rs]: Just an FYI: In https://github.com/mozilla-mobile/android-components/pull/2838, we ultimately got rid of this side-channel information for this and just set first_run whenever that metric doesn't exist in the database.  I think we defer that until we port that PR to glean-core, but just wanted to note it here.

File: glean-core/src/metrics/datetime.rs [hunk 2]
```diff
@@ -108,6 +108,21 @@ impl DatetimeMetric {
         glean.storage().record(&self.meta, &value)
     }
 
+    /// Set the metric to the current date/time, including the local timezone offset.
+    ///
+    /// ## Arguments:
+    ///
+    /// * `glean` - the Glean instance this metric belongs to.
+    pub fn set_now(&self, glean: &Glean) {
```
Comment (mdboom) [inline glean-core/src/metrics/datetime.rs]: I don't feel strongly, but would this be better as `set(&self, glean: &Glean, value: Option<DateTime<FixedOffset>>)` to more closely hew to the Kotlin API?  I'm mainly thinking ahead to when we have a proper public Rust API and we want to document it easily in parallel with the Kotlin API.

File: glean-core/src/metrics/datetime.rs [hunk 3]
```diff
@@ -108,6 +108,21 @@ impl DatetimeMetric {
...
+        if !self.should_record(glean) {
+            return;
+        }
+
+        let value = local_now_with_offset();
+        let value = Metric::Datetime(value, self.time_unit);
```
Comment (mdboom) [inline glean-core/src/metrics/datetime.rs]: Maybe call `self.set(glean, value)` here instead to reduce code duplication?

File: glean-core/src/lib.rs (hunk 1 above)
Comment (badboy) [reply glean-core/src/lib.rs]: Do we have a bug for that in Glean-core?

Comment (mdboom) [reply glean-core/src/lib.rs]: I couldn't find one, so just made one: https://bugzilla.mozilla.org/show_bug.cgi?id=1557366

**Comment by badboy** [comment]
Folded the `set_now` into `set` by requiring an option.

**Pull request merged**